### PR TITLE
Bundle watchdog binaries into an image

### LINKED ISSUE #969: Proposal: Publish Docker image(s) with watchdog binary
## Description

We currently need to install curl / wget to fetch the watchdog / of-watchdog binary for templates. What if we published a set of docker images with each watchdog binary per platform - or one larger docker image with all inside and then used that to speed up the build?

```
FROM openfaas/watchdog:0.9.11 as watchdog
FROM go:1.1.1
COPY --from watchdog /fwatchdog /usr/bin/watchdog
```

Our images would be easier to maintain as per above vs:

```
FROM golang:1.10.4-alpine3.8 as builder

RUN apk --no-cache add curl \
    && echo "Pulling watchdog binary from Github." \
    && curl -sSL https://github.com/openfaas/faas/releases/download/0.9.6/fwatchdog > /usr/bin/fwatchdog \
    && chmod +x /usr/bin/fwatchdog \
    && apk del curl --no-cache
```

What do we think? Perhaps it might help with #957 via @rdimitrov

### PR DESCRIPTION
**Pull request by rgee0**
Signed-off-by: Richard Gee <richard@technologee.co.uk>

## Description
Adds a packager Dockerfile to create an image that contains the fwatchdog binaries.

## Motivation and Context
Fixes #969 

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))
- [x] My issue has received approval from the maintainers or lead with the `design/approved` label

## How Has This Been Tested?
Successfully ran locally.
CI passed

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by alexellis** [comment]
This is definitely useful. What I'm trying to get to is where we have a different tagged image for each architecture/binary. It would mean that rather than using curl we could use multistage builds. 

If all of the binaries are in a single image then that could be quite a big download. I think this can be done on a cloud VM even for arm. Maybe by using scratch?

**Comment by rgee0** [comment]
Currently we cross compile for all platforms in CI - I'm not sure what the reference to a cloud VM pertains to but it feels like you're suggesting we take the image creation out of CI? 

The image is 25MB

File: watchdog/Dockerfile.packager [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FROM openfaas/watchdog:build as build
+FROM scratch
+
+ARG HARDWARE
```
Comment (alexellis) [inline watchdog/Dockerfile.packager]: Minor nit - since Windows isn't "HARDWARE" - platform may work better. This doesn't prevent a merge though.

**Comment by alexellis** [comment]
This is what I had in mind :+1:

**Comment by alexellis** [comment]
Thank you @rgee0

**Pull request merged**